prow/config/branch_protection.go

```diff
@@ -194,7 +200,8 @@ func (o Org) GetRepo(name string) *Repo {
 // Repo holds protection policy overrides for all branches in a repo, as well as specific branch overrides.
 type Repo struct {
 	Policy
-	Branches map[string]Branch `json:"branches,omitempty"`
+	Branches         map[string]Branch `json:"branches,omitempty"`
+	branchExclusions *regexp.Regexp
 }
 
 // GetBranch returns the branch config after merging in any repo policies.
```

Comment:
I do not think `GetBranch` should be returning `nil` -- we should not make the call to `GetBranch` in the first place.

Reply:
Thank you for the review. Do you mean to move the logic for handling branch exclusion into the `protect` command itself instead and let `config` objects provide policies for branches based on the org/repo/branch hierarchy?

Reply:
Yes. When resolving the list of branches we need to know protection rules for, handle the exclusions there so we're not introducing `nil` checks everywhee

Reply:
Updated the PR. Not sure why the file `prow/monitoring/BUILD.bazel` gets updated by running `hack/update-bazel.sh`.

Reply:
That file had some issues... should be fixed if you rebase on `master`